src/Microsoft.DotNet.Interactive/Parsing/DirectiveNode.cs

```diff
@@ -47,39 +47,9 @@ internal DirectiveNode(
 
         if (TryGetDirective(out var directive))
         {
-            foreach (var namedParameter in directive.Parameters)
+            foreach (var diagnostic1 in GetDiagnosticsForMissingParameters(directive, this))
```

Comment:
consider renaming to avoid the `1` suffix :) Would have been nice if C# allowed to omit the var and reuse the `diagnostic` variable